jekyll/index.markdown

```diff
@@ -133,13 +154,229 @@ Signature help often appears right after users finish typing a method, providing
+discoverable and visible.
+
+By default, only implicit rescue hints are displayed. VS Code users can use the following settings to customize inlay
+hint behavior:
+
+```jsonc
+{
+    // Enable all hints
+    "rubyLsp.featuresConfiguration.inlayHint.enableAll": true,
+    // Enable implicit rescue (defaults to true)
+    "rubyLsp.featuresConfiguration.inlayHint.implicitRescue": true,
+    // Enable implicit hash values (omitted hash values)
+    "rubyLsp.featuresConfiguration.inlayHint.implicitHashValue": true
+  }
+```
+
+To configure other editors, see the [initialization options](editors#all-initialization-options).
+
+![Inlay hint demo](images/inlay_hint.gif)
+
+### On type formatting
+
+On type formatting applies changes to the code as the user is typing. For example, the Ruby LSP auto completes the `end` tokens when breaking lines.
+
+{: .note }
+In VS Code, formatting on type is disabled by default. You can enable it with `"editor.formatOnType": true`
+
+![On type formatting demo](images/on_type_formatting.gif)
+
+### Selection range
+
+Selection range (or smart ranges) expands or shrinks a selection based on the code's constructs. In VS Code, this can
+be triggered with `CTRL + SHIFT + LEFT/RIGHT ARROW` to expand/shrink, respectively.
+
+![Selection range demo](images/selection_range.gif)
+
+### Show syntax tree
+
+Show syntax tree displays the Abstract Syntax Tree (AST) for the current Ruby document. This custom feature can either
+show the AST for the entire document or for a selection.
+
+{: .note }
+This feature is not a part of the language server specification. It is a custom feature, which is implemented in the
+Ruby LSP's VS Code extension. Other editors can implement a similar approach to achieve the same functionality
+
+![Show syntax tree demo](images/show_syntax_tree.gif)
+
+### ERB support
+
+The Ruby LSP can process ERB files and handle both the embedded Ruby and the host language portions of the file. For the
+embedded Ruby part, the Ruby LSP responds with all Ruby features you would normally see in regular Ruby files. For
+features for the host language, like HTML, the Ruby LSP delegates the requests to the language service registered to
+handle that file type.
+
+{: .note }
+Request delegation has not yet been formalized as part of the LSP specification. Therefore, this requires custom code
+on the client (editor) side. The Ruby LSP VS Code extension ships with that custom implementation, but other editors
+will need to implement the same to support these features
+
+![ERB features demo](images/erb.gif)
+
+## VS Code features
+
+The following features are all custom made for VS Code.
+
+### Dependencies view
+
+The Ruby LSP contributes a custom dependencies view panel that allows users to navigate the dependencies of their
+projects.
+
+![Dependencies view demo](images/dependencies_view.gif)
+
+### Copilot chat participant
```

Comment:
Are we past considering this experimental? If not, this might belong in the "Experimental Features" list.